components/formatted_markdown_message.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useIntl} from 'react-intl';
+import marked from 'marked';
+
+import {shouldOpenInNewTab, getSiteURL} from 'utils/url';
+
+const TARGET_BLANK_URL_PREFIX = '!';
+
+export class CustomRenderer extends marked.Renderer {
+    private disableLinks: boolean;
+
+    constructor(disableLinks = false) {
+        super();
+        this.disableLinks = disableLinks;
+    }
+
+    link(href: string, title: string, text: string) {
+        const siteURL = getSiteURL();
+        const openInNewTab = shouldOpenInNewTab(href, siteURL);
+
+        if (this.disableLinks) {
+            return text;
+        }
+        if (href[0] === TARGET_BLANK_URL_PREFIX) {
+            return `<a href="${href.substring(1, href.length)}" rel="noopener noreferrer" target="_blank">${text}</a>`;
+        }
+        if (openInNewTab) {
+            return `<a href="${href}" rel="noopener noreferrer" target="_blank">${text}</a>`;
+        }
+        return `<a href="${href}">${text}</a>`;
+    }
+
+    paragraph(text: string) {
+        return text;
+    }
+}
+
+type Props = {
+    defaultMessage?: string;
+    disableLinks?: boolean;
+    id?: string;
+    values?: Record<string, any>;
+}
+
+/**
+ *
+ * Translations component with the same API as react-intl's <FormattedMessage> component except the message string
+ * accepts Markdown.
+ *
+ * @deprecated Use FormattedMessage with {@link https://formatjs.io/docs/react-intl/components/#rich-text-formatting rich text formatting} instead
+ * of including Markdown in translation strings.
+ */
+export default function FormattedMarkdownMessage({
+    id,
+    defaultMessage,
+    values,
+    disableLinks,
+}: Props) {
+    const intl = useIntl();
```

Comment:
Do we have more of such deprecated components ? if so it would be helpful to add such a hook

Reply:
Not that I know of, but I can imagine we'll keep having a few from time to time and the amount of code is pretty low. And if we return `null` from the hook, we can (ab)use it as a "Component" in non-functional components. e.g.

import UseComponentDeprecated from 'components/common/hooks/useComponentDeprecated';

```tsx
export default class SomeDeprecatedComponent() extends React.PureComponent {
  render() {
    return <div>
    {otherContentHere}
    <UseComponentDeprecated />
    </div>
  }
}
```

Reply:
I'd rather keep it as something that shows up in the editor rather than a runtime thing. We already have quite a few warnings in there that make it hard to use the console for debugging, and since we don't have an immediate plan to get rid of these, I wouldn't want to clog up the console with more of these.

I'm mostly wanting to discourage new usage of them at this point. We'll probably end up making an effort to remove usage of FormattedMarkdownMessage in the future, and at that point, we can start making the warnings noisier. I'd want to get it so that standard rich formatting like bolding can be done automatically without having to pass in explicit values for it every time its used though. That requires more changes to how we use the intl library as well.